rsfbclient-rust/src/wire.rs

```diff
@@ -356,6 +356,33 @@ pub fn execute(tr_handle: u32, stmt_handle: u32, input_blr: &[u8], input_data: &
     req.freeze()
 }
 
+/// Execute prepared statement request.
+pub fn execute2(
+    tr_handle: u32,
+    stmt_handle: u32,
+    input_blr: &[u8],
+    input_data: &[u8],
+    output_blr: &[u8],
+) -> Bytes {
+    let mut req =
+        BytesMut::with_capacity(40 + input_blr.len() + input_data.len() + output_blr.len());
+
+    req.put_u32(WireOp::Execute2 as u32);
+    req.put_u32(stmt_handle);
+    req.put_u32(tr_handle);
+
+    req.put_wire_bytes(input_blr);
+    req.put_u32(0); // Input message number
+    req.put_u32(if input_blr.is_empty() { 0 } else { 1 }); // Messages
+
+    req.put_slice(input_data);
+
+    req.put_wire_bytes(output_blr);
+    req.put_u32(0); // Output message number
```

Comment:
i know this is similar to with `fetch` and `execute`, but what is this doing?
Nevermind line 378.

I see it 's part of the [xdr reference previously linked to](https://www.firebirdnews.org/docs/papers/GDSDBProtocol.html#op_execute2) but what are 
`MessageNumber`, `OutPutMessageNumber` and `Messages` actually referring to?

and what is the purpose of the output blr? i see from `xsqlda_to_blr` that it's just a list of ouput types. that seems useless to the server. is it possible to do other things with it?

Reply:
I am not sure, but it seems you can send the input / output data in parts, so this message number is kind of an index, but i don't really know as i never seen it not be 0.

About the output blr, it has the same role as the `coerce` function in the `XSqlDa`of the native version. It tells the server what kind of data the lib is expecting, so it can coerce the data and send it in the right format. For example, we parse all integer types as 64bit numbers, so the output xsqlda tells the server to coerce any present integer type as a 64bit integer.

Reply:
The `Messages` seems to be something on the lines of "how many rows of data are you sending?" so we send 0 when there is no parameters and 1 when there is, but i am not 100% sure.